packit/update_dist_git.py

```diff
@@ -0,0 +1,240 @@
+    # Download the source archives from config_sources
+    for source in config_sources:
+        download(source.url, target / source.path)
+        downloaded.append(source.path)
+    # Download the source archives mentioned in the spec-file
+    # which are not yet downloaded
+    spec = rpm.spec(str(specfile))
+    spec_sources = [source for source in spec.sources if source[2] == 1]
+    spec_sources = [
+        source[0] for source in sorted(spec_sources, key=operator.itemgetter(1))
+    ]
+    spec_sources = [source for source in spec_sources if validators.url(source)]
+    for source_url in spec_sources:
+        file_name = Path(source_url).name
+        source_path = target / file_name
+        if not source_path.exists():
+            download(source_url, source_path)
+            downloaded.append(file_name)
+    return downloaded
+
+
+def update_dist_git(
+    source_git: Path,
+    dist_git: Path,
+    config: Config,
+    pkg_tool: Optional[str] = None,
+    message: Optional[str] = None,
+):
+    """Update a dist-git repo from a source-git repo
+
+    This does the following:
+    - read the package configuration in the source-git repo to tell
+      the upstream_ref
+    - purge tracked files from the dist-git repo to prepare it for
+      the update
+    - copy the content of .distro to the dist-git repo
+    - generate the patch files since upstream_ref and save them to
+      the dist-git repo
+    - update the spec-file with these patches by inserting patch lines
+      after all the sources are defined
+    - download the source archives and upload them to the lookaside
+      cache if a packaging tool is defined
+    - commit all the changes in the dist-git repo if a message is
+      provided
+
+    Arguments:
+        source_git: Path of the source-git repository from which updates
+            are taken.
+        dist_git: Path of the dist-git repository to be updated.
+        config: Packit configuration options from the command line.
+        pkg_tool: Name of the pkg_tool to be used to save source archives
+            in the lookaside cache.
+        message: Commit message of the dist-git commit.
+            Do not commit if this is empty or none.
+    """
+    source_git_repo = git.Repo(source_git)
+    dist_git_repo = git.Repo(dist_git)
+    # make the dist-git working directory empty
+    purge_repo(dist_git_repo, ignore=[".git", ".gitignore", "sources"])
+    # copy files from the .distro directory to dist-git
+    copy_git_repo_dir(source_git_repo, ".distro", dist_git)
+    # get the upstream ref
+    package_config = get_local_package_config(
+        source_git, package_config_path=config.package_config_path
+    )
+    upstream_ref = package_config.upstream_ref
+    # create patch files in dist_git
+    patch_generator = PatchGenerator(source_git_repo)
+    patches = patch_generator.create_patches(
+        upstream_ref, str(dist_git), files_to_ignore=[".distro", ".packit.yaml"]
+    )
```

Comment:
I left this out on purpose, with the idea that downstream packaging files should have a well defined and preferably single place to be stored at, and that should be `.distro`.
Limiting this will simplify tooling code and help with keeping source-git repos consistent across the board (which, for example, could make it easier for proven packagers to integrate source-git repos in their scripts).